### ISSUE TITLE
enable featherlight in editor and info cdata in groups

### ISSUE THREAD

**Issue by ronm123** [thexerteproject/xerteonlinetoolkits#921]
During discussion about accessibility we're wondering about extending the the help available within the wizards for relevant pages e.g. a bit link we currently link to some pdf guides from certain pages but instead linking to an external Xerte LO and opening that in a lightbox.

One example was a guide to adding alt text correctly so rather than just tooltip text we might link to a graphical guide with examples. This would be similar to the pdf links but instead open a single page or multipage LO and ideally open that in a lightbox window.

I tested this with a copy of the graphics and sound page and linked to a xerte lo e.g.

![image](https://user-images.githubusercontent.com/5167146/89559251-5296a400-d80d-11ea-8605-77c1e2dc85e0.png)

and in the xwd I added the following including data-featherlight="iframe" to test if it's already included:
`<info><![CDATA[<b>For full details on how to correctly add alt text <u><a data-featherlight="iframe" href="https://xot.xerte.org.uk/play.php?template_id=151" target="_blank">view this guide</a></u></b>]]></info>`

It seems featherlight isn't included in the editor and I'm not sure how best to include it?

Also ideally it would be better to be able to have info panels within groups but that doesn't seem to work at the moment. e.g. rather than have the info and guide linked at the bottom of the wizard show it at the bottom of the image group. Is that possible?

**Comment by torinfo** [comment]
@ronm123 Featherlight is added to the editor (at least it is included in module/xerte/edithtml.php). So there is probably a different issue that prevents this from working.

**Comment by ronm123** [comment]
@torinfo I've tried to get this working but simply can't see why it isn't! To verify the featherlight css and js was actually loaded I added a background color to the css and an alert to the js and both worked. I also inspected the editor where I've added the link and that shows exactly what should be in place e.g.
`<a data-featherlight="iframe" href="https://xot.xerte.org.uk/play.php?template_id=151">view this guide</a>`

The hyperlink works but just doesn't open in a lightbox. If I set target="_blank" the link opens in a new window and if I remove the target it opens in the same window. If I copy and paste the code above into the source view of a page and play it then click the link it opens in a lightbox. So I'm baffled as to why this doesn't work! Perhaps a challenge for @JohnSmith-LT ;-)

**Comment by torinfo** [comment]
Could this be, because the data is loaded dynamically, i.e. we need to do something to let featherlight do it's work, like we have to trigger mathjax to rescan the contents?

Or call using javascript: $.featherlight({iframe: 'editor.html', iframeMaxWidth: '80%', iframeWidth: 500,
	iframeHeight: 300});

**Comment by ronm123** [comment]
@torinfo I don't know but including the data-featherlight="iframe" should work but isn't.  I was going to explore whether I needed to do anything about the size of the lightbox once the basic opening of the lightbox was working.

**Comment by torinfo** [comment]
@ronm123 What I think is happening is that featherlight is loaded, and on 'document ready' searches all the data-featherlight elements, and does it's thing.  In our scenario, the link is loaded dynamically after the 'document ready' event, and thus, nothing happens. I tried to place 

`$.featherlight({iframe: 'editor.html', iframeMaxWidth: '80%', iframeWidth: 500, iframeHeight: 300});`

in the onclick event of your link example, but that doesn't seem to work. Not even in an anonymous function. So this probably requires extra code to xenith or the editor to facilitate this.

@JohnSmith-LT Do you have an idea on how to get this working?

**Comment by JohnSmith-LT** [comment]
Is it the same version? I'm sure we had an old version and a newer version of featherlight on the site. If you look at the network tab (filter featherlight) for the editor and for an LO then check both versions are the same as I think iframe was added later...

**Comment by JohnSmith-LT** [comment]
@ronm123 regarding the infopanel thing above - if you open the editor and add the Variables option then you'll see that Fay has added instructions within the group.

**Comment by JohnSmith-LT** [comment]
Yes, the editor uses a different version, we should just get rid of that old version on the site...

![image](https://user-images.githubusercontent.com/3907362/89780276-06977800-db09-11ea-8c0f-0ff7e28b86b1.png)

![image](https://user-images.githubusercontent.com/3907362/89780346-2d55ae80-db09-11ea-8938-6762b6c4e3de.png)

**Comment by ronm123** [comment]
@JohnSmith-LT thanks for the tip re variables group and info link. I was trying to use <info>...</info> within the group rather than info="..." within the group.

regarding the different versions of featherlight - so was the old version in the editor because we use it somewhere and if we change it to the same version used elsewhere will it break anything? What file exactly does the include of featherlight into edithtml.php happen?

**Comment by ronm123** [comment]
@JohnSmith-LT @torinfo I found a featherlight folder at \editor\js\vendor\featherlight\ so replaced the files in there from \modules\xerte\parent_templates\Nottingham\common_html5\js\featherlight\ and if then use the console to inspect the versions I can see the newer css but not the newer js. What am I missing?

**Comment by JohnSmith-LT** [comment]
Not sure it is ever used, maybe @torinfo may know but anyway, i've just changed the featherlight file and edited the Variables doc link to your link above and it now works, albeit that the min dimensions haven't probably been set yet...

![image](https://user-images.githubusercontent.com/3907362/89787299-1e74f900-db15-11ea-8b79-bee95d8cab70.png)

**Comment by JohnSmith-LT** [comment]
Sorry @ronm123 just saw the previous message.. you the edithtml.php file is loading the featherlight.js file NOT the featherlight.min.js file so that needs editing at /modules/xerte/edithtml.php (and probably in the other places too for bootstrap etc...)

**Comment by torinfo** [comment]
@ronm123 @JohnSmith-LT I replaced the geatherlight instances in all 3 edithtml.php files and removed the old featherlight folder.

Not committed yet. Do you want me to commit?

**Comment by ronm123** [comment]
@torinfo yes please commit unless @JohnSmith-LT has any reasons why you shouldn't?

**Comment by ronm123** [comment]
BTW the easiest way to fix the sizing would be to include `.featherlight iframe {height: 90vh;width: 90vw;}` somewhere.

**Comment by torinfo** [comment]
@ronm123 @JohnSmith-LT Committed the necessary changes to edithtml.php (3x) to use the featherlight used by xenith, and removed the one installed in the editor folder. I did NOT commit any changes to the textGrapghics page (but I did verify that adding a onclick event works, but as John said, your original solution should now work as well.

**Comment by JohnSmith-LT** [comment]
Sounds good to me. Just to confirm that I only tested inserting that link <a data-featherlight="iframe" href="https://xot.xerte.org.uk/play.php?template_id=151">view this guide</a> into the info field used by the Variables group by editing the html using dev tools in Chrome and featherlight automatically picks that up without having to do any onclick event etc...

**Comment by torinfo** [comment]
@JohnSmith-LT Yes. I understood that. Just to prevent double work. 

@JohnSmith-LT @ronm123 
Who is going to add the info field and the change to the css file?

**Comment by ronm123** [comment]
@JohnSmith-LT this was just a proof of concept test I don't have an info field at add/commit at the moment just one I'm testing this with which I'll do locally. So I'll do that when we have a real world example and I have one in mind.

What css file should I add the`.featherlight iframe {height: 90vh;width: 90vw;}` to? You mentioned the editor css but what file exactly?

**Comment by torinfo** [comment]
@ronm123 The css files used in the editor are specified in editor/js/vendor/ckeditor/config.js.

You could add a css file in that folder and include that in the config.js. The xerteedtor.css in that folder is almost what you want, but that css file is only included in case a them is used, different than the default theme. (See config.js). Let me know if you wnat me to add that file.

**Comment by JohnSmith-LT** [comment]
If you need me to look at it @ronm123 then let me know and I can do tonight.

**Comment by ronm123** [comment]
@torinfo @JohnSmith-LT it seems overkill to create a whole new style sheet just for this little bit of css! I checked what existing sheets are loaded with the editor open e.g.
![image](https://user-images.githubusercontent.com/5167146/89809494-889c9680-db33-11ea-8683-e94d08fa9ebd.png)
and there's one named xerte_buttons.css which is obviously one of ours and not likely to be overwritten and with this bit of css in that it works and the iframe fills the editor window with those responsive dimensions. Shall we just use that and if so I can commit?

**Comment by JohnSmith-LT** [comment]
Seems to me like the /editor/css/complex.css (though strangely named) appears to be the generic css for the editor as it has all sorts of stuff in there from structure to loading spinner definitions?

**Comment by ronm123** [comment]
@JohnSmith-LT I can’t check at the moment but is that ours or part of the core editor code? What’s needed works in xerte_buttons.css but I know that name and primary purpose of that perhaps means complex.css would be better if that’s one of ours too?

**Comment by torinfo** [comment]
@ronm123 @JohnSmith-LT Yes, that's ours too.

**Comment by ronm123** [comment]
I just committed `.featherlight iframe {height: 90vh;width: 90vw;}` to complex.css so will close this issue.
